jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java

```diff
@@ -1015,23 +1015,14 @@ private ServletResourceService(ServletContextHandler servletContextHandler)
         }
 
         @Override
-        public String getWelcomeTarget(Request coreRequest)
+        public String getWelcomeTarget(HttpContent content, Request coreRequest)
         {
             String[] welcomes = _servletContextHandler.getWelcomeFiles();
             if (welcomes == null)
                 return null;
-
-            HttpServletRequest request = getServletRequest(coreRequest);
             String pathInContext = Request.getPathInContext(coreRequest);
-            String includedServletPath = (String)request.getAttribute(RequestDispatcher.INCLUDE_SERVLET_PATH);
-            String requestTarget;
-            if (includedServletPath != null)
-                requestTarget = getIncludedPathInContext(request, includedServletPath, isPathInfoOnly());
-            else
-                requestTarget = isPathInfoOnly() ? request.getPathInfo() : pathInContext;
-
             String welcomeTarget = null;
-            Resource base = _baseResource.resolve(requestTarget);
+            Resource base = content.getResource();
```

Comment:
Do we need more null checks on content before using it? Both here and in `ResourceService.doGet()`?